File: data-generators/RecordsToFile.java
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Random;

public class RecordsToFile {

    private static int getRandomIndex(int size) {
        return new Random().nextInt(size);
    }

    private static double getRandomPrice() {
        return 0.5 + new Random().nextDouble()*(1.8-0.5);
    }

    public static void main(String[] args) {
        if (args.length!=2) {
            System.out.println("Usage: java DataGenerator <output_file> <record_count>");
            return;
        }

        String[] productNames = {"Apple", "Banana", "Lemon", "Cherry", "Melon", "Peach", "Grapefruit"};
        int productNamesSize = productNames.length;

        String outputDir = "../records/";
        String outputFilePath = outputDir+args[0];

        int recordCount = Integer.parseInt(args[1]);

        try (PrintWriter writer = new PrintWriter(new FileWriter(outputFilePath))) {

            for (int i=0; i<recordCount; i++) {
                int productNameIndex = getRandomIndex(productNamesSize);
                double price = getRandomPrice();

                String json = String.format("{\"id\": %d, \"name\": \"%s\", \"price\": %.2f}\n", i, productNames[productNameIndex], price);
                writer.write(json);
                writer.flush();
            }
        } catch (IOException e) {
            System.out.println("Error writing to file: " + e.getMessage());
            return;
        }
    }
}

File: data-generators/recordsToFile.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

int getRandomIndex(int size) {
    return rand()%size;
}

double getRandomPrice() {
    return 0.5 + ((double)rand()/RAND_MAX) * (1.8-0.5);
}

int main(int argc, char* argv[]) {
    if (argc!=3) {
        printf("Usage: %s <output_file> <record_count>\n", argv[0]);
        return 1;
    }

    const char* productNames[] = {"Apple", "Banana", "Lemon", "Cherry", "Melon", "Peach", "Grapefruit"};
    int productNamesSize = sizeof(productNames)/sizeof(productNames[0]);

    const char* output_dir = "../records/";
    char output_file_path[100];
    snprintf(output_file_path, sizeof(output_file_path), "%s%s", output_dir, argv[1]);

    int record_count = atoi(argv[2]);

    srand(time(NULL));

    FILE* file = fopen(output_file_path, "w");
    if (!file) {
        printf("Error opening file: %s\n", output_file_path);
        return 1;
    }

    for (int i=0; i<record_count; i++) {
        int productNameIndex = getRandomIndex(productNamesSize);
        double price = getRandomPrice();

        char jsonBuffer[128];
        snprintf(jsonBuffer, sizeof(jsonBuffer), "{\"id\": %d, \"name\": \"%s\", \"price\": %.2f}\n", i, productNames[productNameIndex], price);

        fputs(jsonBuffer, file);
        fflush(file);
    }
    fclose(file);

    return 0;
}
